### DISCUSSION TITLE
Updating from version 4.2.1 to 4.2.4

### DISCUSSION THREAD

**Discussion by 12345ieee** [oracle-terraform-modules/terraform-oci-oke#534] [General]
Hello, we recently updated from version 4.2.1 to 4.2.4 on some clusters.

The VCN changes introduced (I think in v4.2.3) mean that it's needed to run:
`terraform state mv module.oke.module.vcn module.oke.module.vcn[0]`
to avoid mass cluster recreation.

**Comment by hyder** [comment]
Hi @12345ieee,

Apologies. I missed this discussion. Can you please elaborate?

**Comment by 12345ieee** [comment]
Replicating the issue is easy:
* create a cluster with module version 4.2.1
* update the module version to 4.2.4 (4.2.3 should suffice, but I didn't want to track the change down)
* fix the 2-3 parameters that changed in the module call
* run `terraform plan`, you will see a ton of deletion/creation attempts of vcn-related resources

this happens because `module.oke.module.vcn` used to be a single resource, it is now a list so terraform wants to recreate the vcn and its dependencies.

The actual cloud resources that the module wants to make didn't change though, so just renaming the resource fixes the plan delta.

This migration pain could be avoided if the module included a `moved` block, but because we spent some time figuring it out I wanted to let the community know, things like these should be written in the Release page under "backwards incompatible changes" IMHO.
